src/core/ext/transport/chttp2/client/secure/secure_channel_create.c

```diff
@@ -60,69 +60,57 @@ static void client_channel_factory_unref(
 static grpc_subchannel_args *get_secure_naming_subchannel_args(
```

Comment:
Now that we're doing this for all channels, not just LB channels, this should *always* succeed.  In other words, we should change client_channel_factory_create_subchannel() such that if the call to this function fails, then the subchannel creation fails.  Otherwise, we will wind up simply creation a channel without a security connector, in which case we would fail to perform security actions on the subchannel, which seems dangerous.